### PR TITLE
JARVIS-620 - Add support on ThumbnailManager for a load listener on reps

### PR DESCRIPTION
**Pull request by rgcorrea**
In order to allow showing error messages when loading representations, this
patch introduces the possibility of setting a listener on representation
load request, so that the calling component can provide special handling
for loading error, or success.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @rgcorrea has signed the CLA. Thanks for the pull request!

File: box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java [hunk 1]
```diff
@@ -530,18 +541,62 @@ public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
...
+    private static class ViewData {
+        private ImageLoadListener mListener;
+        private String mImageType;
+
+        ViewData(String type, ImageLoadListener listener) {
+            mImageType = type;
+            mListener = listener;
+        }
+
+        static ImageLoadListener getImageLoadListener(ImageView view) {
```
Comment (jimmy-holzer-box) [inline box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java]: Is there any advantage to have those 2 static function here (getImageLoadListener and getImageType)? Why not just put them (not static) in ThumbnailManager.java?

File: box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java [hunk 2]
```diff
@@ -530,18 +541,62 @@ public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
...
+            if(view != null) {
+                ViewData data = (ViewData) view.getTag();
+                if(data != null) {
+                    return data.mListener;
+                }
+            }
+            return null;
```
Comment (jimmy-holzer-box) [inline box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java]: Should we use the null object pattern so the caller never have to worry if this function return null?

File: box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java (hunk 1 above)
Comment (rgcorrea) [reply box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java]: As the data is owned by ViewData class, I thought it make more sense to have getters in there.

File: box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java (hunk 2 above)
Comment (rgcorrea) [reply box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java]: I like that.... let me add this. Thanks!

File: box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java [hunk 3]
```diff
@@ -530,18 +535,66 @@ public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
...
+
+        static ImageLoadListener getImageLoadListener(ImageView view) {
+            if(view != null) {
+                ViewData data = (ViewData) view.getTag();
+                if(data != null) {
+                    return data.mListener;
+                }
+            }
+            // Dummy listener, so the caller does not need to worry about null checks (Null object pattern)
+            return new ImageLoadListener() {
```
Comment (jimmy-holzer-box) [inline box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java]: Not a big difference but usually they do:

private static ViewData NULL_OBJECT = new ImageLoadListener() {
  
            
  Write  Preview

Not a big difference but usually they do:
```java
private static final ViewData NULL_OBJECT = new ImageLoadListener() {
  @Override
 public void onSuccess() {}
 
  @Override
   public void onError() {}
};
```
and then 
```java
return NULL_OBJECT;

Comment (rgcorrea) [reply box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java]: done

**Pull request merged**